### ISSUE TITLE
Thumbs doesn't work properly

### ISSUE THREAD

**Issue by RodrigoTomeES** [nolimits4web/swiper#4748]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://codesandbox.io/s/bug-swiper-qjp50

### Bug description

Hi, 

I have created a codesandbox with an example of the error that you can find in this issue but it is basically the following:

At first when you navigate through the slider it skipes some thumbs, and when you go through the slider completely and go back to the beginning, the active thumb stop working.

https://user-images.githubusercontent.com/13077343/125082581-b1bd0280-e0c7-11eb-9e04-83603d38b8bd.mp4

### Expected Behavior

At first when you navigate through the slider it skipes some thumbs, and when you go through the slider completely and go back to the beginning, the active thumb stop working.

### Actual Behavior

I hope the thumbs change to active as I navigate the slider.

### Swiper version

6.7.5

### Platform/Target and Browser Versions

Windows 10 / Firefox

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by xaja** [comment]
Have the same in Vue. 
so this is a core bug..

**Comment by RodrigoTomeES** [comment]
Could any collaborator change the label? Thanks!

**Comment by RodrigoTomeES** [comment]
Hi, I have updated the example to use Swiper v7 but the bug still persists. Has anyone the same issue?

**Comment by RodrigoTomeES** [comment]
@vltansky thanks!, Have you been able to see the issue? Do you have the same problem?

**Comment by pfasang** [comment]
We also discovered multiple bugs with thumb nav. Could someone check this functionality?

**Comment by RodrigoTomeES** [comment]
Hi, the bug is still in the latest version 8.0.7, can any one take a look about it? please! 😰 I update the example in codesandbox

**Comment by RodrigoTomeES** [comment]
@xaja @pfasang @kissshot, the user @Richard-08 show me a workaround in the issue #5568 -> https://codesandbox.io/s/bug-swiper-forked-yr1xyj?file=/src/Components/Slider/Slider.js

You only need to move at the same time the thumbs and the slider.

![imagen](https://user-images.githubusercontent.com/13077343/159724197-c40088cc-d5ae-4141-ad70-17a3df3e575b.png)

**Comment by xaja** [comment]
yes, but it is "dirty fix", bug is still exist 
and behaviour of thumbs is not the same — after this "fix" you always have active thumb on first place (if possible)
here is demo https://swiperjs.com/demos/300-thumbs-gallery/core you can compare behaviours of thumb transitions

**Comment by RodrigoTomeES** [comment]
@xaja You are right but at least the slider is usable with this workaround

**Comment by xaja** [comment]
Yes, that is true, too

**Comment by solankeys** [comment]
i am facing the same issue. i am using 8.2.2 version. why is not fixed yet

**Comment by onderakbulut** [comment]
Not working on vue 3 vite. can u check demo page?

**Comment by bora89** [comment]
Yes, demo page for Vue 3 thumbnail one is not working at all. 
![CleanShot 2023-05-03 at 13 04 55@2x](https://user-images.githubusercontent.com/1118710/235887863-896b7d03-527f-48a3-8af5-2e3612e8c04d.png)

**Comment by bora89** [comment]
> Not working on vue 3 vite. can u check demo page?

[Ok, I have managed to fix Vue 3 thumbnails demo](https://codesandbox.io/p/sandbox/swiper-vue-thumbs-gallery-forked-y5qrh6)

**Comment by nolimits4web** [comment]
Issue is closed because of outdated/irrelevant/not actual/needed more information/inactivity.

If this issue is still actual and reproducible for latest version of Swiper, please create new issue and fill the issue template correctly:
* Clearly describe the issue including steps to reproduce when it is a bug.
* Make sure you fill in the earliest version that you know has the issue.
* Provide live link or JSFiddle/Codepen or website with issue

**Comment by Spirik** [comment]
Faced the same issue on my Next.js project. Curiously enough the same code worked perfectly fine under simple react (not Next.js based) app.

During my investigation, I've noticed that resizing window helps: once the classes fail to be set correctly on the Thumbs slider, just resize browser window a little, and that will trigger repaint of the sliders and classes will be applied correctly.

So I've added to my Thumbs slider the following event handler `onSlideChange={handleSlideChange}` which simply dispatches a window resize event (and that forces the slider to be redrawn correctly):

```jsx
function handleSlideChange() {
    window.dispatchEvent(new Event('resize'));
}
```

It may be a good idea to wrap this code in `setTimeout` if the above direct call won't help:

```jsx
function handleSlideChange() {
    setTimeout(() => {
        window.dispatchEvent(new Event('resize'));
    }, 0);
}
```

Naturally I've thought that it would be better to fire some `swiper.update()` or `swiper.updateSlidesClasses()` methods instead. But that did nothing. So I've settled with dispatching resize event workaround for now.

Just to note, this solution does not require changing behavior of the Navigation buttons in any way (i.e. writing your own button press handlers): arrows work, keyboard navigation works (if enabled).

Hopefully, someone will find this workaround useful.
